src/Symfony/Component/ClassLoader/ClassCollectionLoader.php

```diff
@@ -116,8 +116,8 @@ public static function load($classes, $cacheDir, $name, $autoReload, $adaptive =
         }
 
         // cache the core classes
-        if (!is_dir(dirname($cache))) {
-            mkdir(dirname($cache), 0777, true);
+        if (!file_exists($cacheDir) && !@mkdir($cacheDir, 0777, true) && !is_dir($cacheDir)) {
+            throw new \RuntimeException('Class Collection Loader was not able to create a folder: '.$cacheDir);
```

Comment:
I would call it a 'directory' instead and make sure exceptions end with a full stop, same goes for the other exceptions.